public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -11,12 +12,60 @@ interface DimensionFieldsProps extends AzureQueryEditorFieldProps {
   dimensionOptions: AzureMonitorOption[];
 }
 
-const DimensionFields: React.FC<DimensionFieldsProps> = ({ query, dimensionOptions, onQueryChange }) => {
+interface DimensionLabels {
+  [key: string]: Set<string>;
+}
+
+const DimensionFields: React.FC<DimensionFieldsProps> = ({ data, query, dimensionOptions, onQueryChange }) => {
   const dimensionFilters = useMemo(
     () => query.azureMonitor?.dimensionFilters ?? [],
     [query.azureMonitor?.dimensionFilters]
   );
 
+  const [dimensionLabels, setDimensionLabels] = useState<DimensionLabels>({});
+
+  useEffect(() => {
+    let labelsObj: DimensionLabels = {};
+    if (data?.series?.length) {
+      const series: DataFrame[] = data.series.flat();
+      const fields = series.flatMap((item) => item.fields);
+      const labels = fields
+        .map((item) => item.labels)
+        .flat()
+        .filter((item) => item!);
+      for (const label of labels) {
+        if (label) {
+          for (const [dimension, value] of Object.entries(label)) {
+            if (labelsObj[dimension]) {
+              labelsObj[dimension].add(value);
+            } else {
+              labelsObj[dimension] = new Set([value]);
+            }
+          }
+        }
+      }
+    }
+    setDimensionLabels((prevLabels) => ({ ...labelsObj, ...prevLabels }));
```

Comment:
If I understand correctly, it extracts the labels from all the returned series and put them in a multi select filter? I'm not sure this is correct. data.series returns labels for all queries in a panel - not only for the current query row. To be sure you use labels for the current query, you need to check the frame's refId against the row that is currently being rendered in the editor. 

I think this hook is a bit complicated. Would be nice if you could add some comments on what it does. Also, make sure all args in the arrow functions are properly named. For example `fields.map((item) => item.labels)` could be `fields.map((field) => field.labels)` to make it easier to read the code. 

You can also consider 1) moving this to a custom hook or 2) moving the multi-select filter to a separate component that would also contain this hook.

Reply:
Great catch on needing to check the refId, will update this.